### PR TITLE
Refactor Azure Spring projects structure

### PR DESCRIPTION
**Pull request by saragluna**
This PR is refactoring the Azure Spring project structures.

Remove azure-spring-boot (the autoconfiguration module), move the code to azure-spring-cloud-autoconfigure
Remove azure-spring-boot-core, move the code to azure-spring-cloud-context
Make a separate module azure-spring-cloud-resourcemanager

### CODE REVIEW

**Comment by saragluna** [comment]
@stliu, @yiliuTo please help review this PR.

**Review by yiliuTo — APPROVED** [review]
LGTM

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhub/AzureEventHubAutoConfiguration.java [hunk 1]
```diff
@@ -6,16 +6,14 @@
...
 import com.azure.messaging.eventhubs.EventHubConsumerAsyncClient;
 import com.azure.resourcemanager.AzureResourceManager;
-import com.azure.spring.cloud.autoconfigure.context.AzureContextAutoConfiguration;
-import com.azure.spring.cloud.autoconfigure.context.AzureContextProperties;
-import com.azure.spring.cloud.context.core.api.EnvironmentProvider;
+import com.azure.spring.cloud.autoconfigure.context.AzureResourceManagerAutoConfiguration;
+import com.azure.spring.cloud.context.core.api.AzureResourceMetadata;
 import com.azure.spring.cloud.context.core.impl.EventHubNamespaceManager;
 import com.azure.spring.cloud.context.core.impl.StorageAccountManager;
 import com.azure.spring.cloud.context.core.storage.StorageConnectionStringProvider;
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/eventhub/AzureEventHubAutoConfiguration.java:13]: StorageConnectionStringProvider class could be moved to package com.azure.spring.cloud.autoconfigure.storage of lib azure-spring-cloud-autoconfigure as well.

**Comment by saragluna** [comment]
Will create another PR to address:
- Whether to move `StorageConnectionStringProvider` to azure-spring-cloud-context
- Should add `subscriptioId` and `tenantId` to `AzureResourceMetadata`

File: sdk/spring/azure-spring-cloud-resourcemanager/src/main/java/com/azure/spring/cloud/context/core/api/AzureResourceMetadata.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+package com.azure.spring.cloud.context.core.api;
+
+/**
+ * Metadata defining an Azure resource.
+ */
+public class AzureResourceMetadata {
+
+    private String region;
+    private String resourceGroup;
+    private boolean autoCreateResources;
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-resourcemanager/src/main/java/com/azure/spring/cloud/context/core/api/AzureResourceMetadata.java:13]: The property of tetantId and subscriptionId could be put here as well.

**Pull request merged**